apps/ledger-live-desktop/src/renderer/hooks/useRecoverRestoreOnboarding.ts

```diff
@@ -0,0 +1,59 @@
+import { useCallback, useEffect, useState } from "react";
+import getUser from "~/helpers/user";
+import { getStoreValue } from "~/renderer/store";
+import { useFeature } from "@ledgerhq/live-common/featureFlags/index";
+import { saveSettings } from "~/renderer/actions/settings";
+import { useDispatch, useSelector } from "react-redux";
+import { useLocation } from "react-router-dom";
+import { hasCompletedOnboardingSelector } from "~/renderer/reducers/settings";
+import { useCustomPath } from "@ledgerhq/live-common/hooks/recoverFeatureFlag";
+
+const ONBOARDED_VIA_RECOVER_RESTORE_USER_PREFIX = "ONBOARDED_VIA_RECOVER_RESTORE_USER_";
+
+export const useRecoverRestoreOnboarding = (seedPathStatus?: string) => {
+  const { pathname } = useLocation();
+  const dispatch = useDispatch();
+  const recoverServices = useFeature("protectServicesDesktop");
+  const recoverStoreId = recoverServices?.params?.protectId ?? "";
+  const hasCompletedOnboarding = useSelector(hasCompletedOnboardingSelector);
+  const recoverRestoreStaxPath = useCustomPath(recoverServices, "restore", "lld-onboarding-24");
+
+  const [onboardedViaRecoverRestore, setOnboardedViaRecoverRestore] = useState<boolean>();
```

Comment:
It can be undefined ?

Reply:
yes, for more info, please see [here](https://github.com/LedgerHQ/ledger-live/pull/6638#discussion_r1570507417).